code/datums/components/food/decomposition.dm

```diff
@@ -0,0 +1,88 @@
+//"Don't leave food on the floor, that's how we get ants"
+/datum/component/decomposition
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	///How decomposed a specific food item is. Uses delta_time.
+	var/decomposition_level = 0
+	///Makes sure food only starts decomposing if a player's EVER picked it up before
+	var/handled = FALSE
+	///Used to stop food in someone's hand & in storage slots from decomposing.
+	var/protected = FALSE
+
+/datum/component/decomposition/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_MOVABLE_MOVED, .proc/handle_movement)
+	RegisterSignal(parent, list(
+        COMSIG_ITEM_PICKUP, //person picks up an item
+        COMSIG_STORAGE_ENTERED), //Object enters a storage object (boxes, etc.)
+        .proc/picked_up)
+	RegisterSignal(parent, list(
+        COMSIG_ITEM_DROPPED, //Object is dropped anywhere
+        COMSIG_STORAGE_EXITED), //Object exits a storage object (boxes, etc)
+        .proc/dropped)
+
+/datum/component/decomposition/UnregisterFromParent()
+	UnregisterSignal(parent, COMSIG_MOVABLE_MOVED, .proc/handle_movement)
+	UnregisterSignal(parent, list(
+        COMSIG_ITEM_PICKUP,
+        COMSIG_STORAGE_ENTERED),
+        .proc/picked_up)
+	UnregisterSignal(parent, list(
+        COMSIG_ITEM_DROPPED,
+        COMSIG_STORAGE_EXITED),
+        .proc/dropped)
```

Comment:
Ditto here. Oh and you don't need to specify the proc to unregister from, you can just say UnregisterSignal(thing to unregister from, signals to unregister)